server-side/controllers/user.controller.js

```diff
@@ -30,40 +31,39 @@ export const getUserById = async (req, res,next) => {
   }
 };
 
-
-export const addUser = async (req, res,next) => {
+export const addUser = async (req, res) => {
   try {
-    if (req.file ) {
-     req.body.profileImage=req.file.originalname;
-     req.body.password= await bcrypt.hash(req.body.password, 10);
+    const { name, email, password } = req.body;
+    if (!name || !email || !password) {
+      return res.status(400).send('Missing required fields');
+    }
+    if (req.file ) 
+     req.body.profileImage  =req.file.originalname;
```

Comment:
don't change the req.body, if you need to add something, please create new object